src/iterator.rs

```diff
@@ -271,67 +279,116 @@ where
 
         bail!("Bug: Internal node has less leaves than expected.");
     }
+
+    /// Internal helper function used in the `Stream` impl.
+    ///
+    /// Gets the current node key and the result of trying to get it out of the tree, or returns
+    /// `Poll::Pending` if necessary to fulfill the request.
+    fn poll_get_current_node(&mut self, cx: &mut Context<'_>) -> Poll<(NodeKey, Result<Node>)>
+    where
+        R: 'static + Send + Sync,
+    {
+        // If we have not yet created a future for polling, make one and store it
+        if self.future.is_none() {
+            // Compute the appropriate node key depending on whether we are at the root or not
+            let node_key = if self.parent_stack.is_empty() {
+                // Root node
+                NodeKey::new_empty_path(self.version)
+            } else {
+                // Next child node
+                let last_visited_node_info = self
+                    .parent_stack
+                    .last()
+                    .expect("We have checked that self.parent_stack is not empty.");
+                let child_index =
+                    Nibble::from(last_visited_node_info.next_child_to_visit.trailing_zeros() as u8);
+                last_visited_node_info.node_key.gen_child_node_key(
+                    last_visited_node_info
+                        .node
+                        .child(child_index)
+                        .expect("Child should exist.")
+                        .version,
+                    child_index,
+                )
+            };
+
+            // Set the future so we can poll it
+            let reader = self.reader.clone();
+
+            self.future = Some(Box::pin(async move {
+                let result = crate::reader::get_node_async(&*reader, &node_key).await;
+                (node_key, result)
+            }));
+        }
+
+        // Get the future
+        let mut future = std::mem::take(&mut self.future).unwrap();
+
+        match future.as_mut().poll(cx) {
+            Poll::Ready(result) => {
+                // In this case, `self.future` is now `None`
+                Poll::Ready(result)
+            }
+            Poll::Pending => {
+                self.future = Some(future); // Put the future back into `self.future`
+                Poll::Pending
+            }
+        }
+    }
 }
 
-impl<R> Iterator for JellyfishMerkleIterator<R>
+impl<R> futures::Stream for JellyfishMerkleStream<R>
 where
-    R: TreeReader,
+    R: TreeReader + Send + Sync + 'static,
 {
-    type Item = Result<(KeyHash, OwnedValue)>;
+    type Item = Result<(KeyHash, ValueHash)>;
```

Comment:
Why does this return a `ValueHash` instead of the full `OwnedValue` now?

Reply:
Oops -  fixed!